src/main/asciidoc/_chapters/upgrading.adoc

```diff
@@ -799,6 +806,21 @@ To upgrade an existing HBase 1.x cluster, you should:
 * Upgrade RegionServers
 * (Eventually) Upgrade Clients
 
+[[upgrade1.7.x]]
+=== Upgrading to 1.7.1+
+HBase release 1.7.0 introduced an incompatible table metadata serialization format that broke the
+minor release compatibility guarantees. The issue was reported in https://issues.apache.org/jira/browse/HBASE-26021[HBASE-26021] and the problematic serialization patch was reverted in
+HBase 1.7.1. Some important notes about 1.7.x upgrades below.
+
+* If you are considering an upgrade to 1.7.x version, skip 1.7.0 completely and upgrade to 1.7.1+
+version. 1.7.0 was withdrawn and removed from the Apache sites.
+* If you already installed a 1.7.0 cluster from scratch and are looking to migrate to 1.7.1+, you
+cannot follow the regular rolling upgrade procedures due to broken compatibility contracts.
+Instead shutdown the  cluster and reboot with 1.7.1+ binaries. Newer versions detect any existing
```

Comment:
tiny nit: double space between "the" and "cluster"

Reply:
Ack.

Reply:
For 1.7.0 to 1.7.1, do clients have to shutdown the cluster? I thought it was about restarting master only.

Reply:
In 1.7.0, client expects TableState (faulty serialization pb message) in GetTableStateResponse. Given that serialization has been undone (as we do not want to expose TableState anymore), client is not compatible (certain RPCs) with restarted Master. So it is safe to shutdown, update binaries everywhere and reboot the cluster. 

I _think_ this is ok given it is impossible that any serious user has upgraded an existing cluster to 1.7.0. Its only about new installations.